cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitor.java

```diff
@@ -603,7 +688,8 @@ private void populateLoad(Cluster kafkaCluster,
         boolean isOffline = Arrays.stream(partitionInfo.offlineReplicas())
                                   .anyMatch(offlineReplica -> offlineReplica.id() == replica.id());
 
-        clusterModel.createReplica(rack, replica.id(), tp, index, isLeader, isOffline);
+        String logdir = replicaPlacementInfo == null ? null : replicaPlacementInfo.get(tp).get(replica.id());
```

Comment:
if `replicaPlacementInfo` is not `null`, but empty or has an in progress logdir move for the given `tp`, then this `logDir` will be `null` -- is this intentional?

Reply:
yes, there will be one case that a lookup here will miss.
1) [replica is on dead broker](https://github.com/linkedin/cruise-control/blob/8ba6410e24d465b873e2c5f999bb220426654076/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitor.java#L574), in this case we ignore this replica's logdir info. This is ok since (1) internally we do not heal offline replica with intra-broker operation,  the two intra-broker goals does not consider these 
    replicas (2)externally we do not [report dead disk's utilization ](https://github.com/linkedin/cruise-control/blob/8ba6410e24d465b873e2c5f999bb220426654076/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/Broker.java#L655)

If anything else goes wrong when retrieving logdir inform, we already [panic before reach here](https://github.com/linkedin/cruise-control/blob/8ba6410e24d465b873e2c5f999bb220426654076/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/LoadMonitor.java#L577).

For replica which is currently moving, we will get its current logdir instead of its future logdir from lookup here.

Reply:
Can we document these expectations / assumptions on JavaDoc?